src/content/docs/dev/general-message-passing/sui/gmp-tutorial.mdx

```diff
@@ -0,0 +1,918 @@
+
+```toml
+[dependencies]
+AxelarGateway = { git = "https://github.com/axelarnetwork/axelar-cgp-sui.git", rev = "b35227a82f7ca6f941a7e729e1a3b3d48a64b982", subdir = "move/axelar_gateway" }
+GasService = { git = "https://github.com/axelarnetwork/axelar-cgp-sui.git", rev = "b35227a82f7ca6f941a7e729e1a3b3d48a64b982", subdir = "move/gas_service" }
+RelayerDiscovery = { git = "https://github.com/axelarnetwork/axelar-cgp-sui.git", rev = "b35227a82f7ca6f941a7e729e1a3b3d48a64b982", subdir = "move/relayer_discovery" }
+Utils = { git = "https://github.com/axelarnetwork/axelar-cgp-sui.git", rev = "b35227a82f7ca6f941a7e729e1a3b3d48a64b982", subdir = "move/utils" }
+VersionControl = { git = "https://github.com/axelarnetwork/axelar-cgp-sui.git", rev = "b35227a82f7ca6f941a7e729e1a3b3d48a64b982", subdir = "move/version_control" }
+
+
+<Callout emoji="💡">
+  Note: This is a temporary workaround for adding dependencies, this will be updated for a more permanent solution.
+</Callout>
+```
+
+### Imports
+
+Before writing up the logic of your package you must import the necessary modules that you will need throughout the package. 
+
+From the [Axelar Gateway's](/dev/general-message-passing/sui/gmp-example/#sui-gateway) [channel mod](https://github.com/axelarnetwork/axelar-cgp-sui/blob/main/move/axelar_gateway/sources/channel.move) you will need the `Channel` and `ApprovedMessage` objects. The gateway's [gateway module](https://github.com/axelarnetwork/axelar-cgp-sui/blob/main/move/axelar_gateway/sources/gateway.move) will use the Gateway object.
+
+```rust
+use axelar_gateway::channel::{Self, Channel, ApprovedMessage};
+use axelar_gateway::gateway::{Self, Gateway};
+```
+From the [Axelar Gas Service](/dev/general-message-passing/sui/gmp-example/#sui-gas-service) you will use the Gas Service object.
+
+```rust
+use gas_service::gas_service::GasService;
+```
+
+From the [Relayer Discovery](https://github.com/axelarnetwork/axelar-cgp-sui/tree/main/move/relayer_discovery) you will use the Relayer Discovery object and Discovery mechanic
+
+```rust
+use relayer_discovery::discovery::RelayerDiscovery;
+use relayer_discovery::transaction;
+```
+
+From standard `move` library helpers you can use the following
+
+```rust
+use std::ascii::{Self, String};
+use std::type_name;
+```
+
+Lastly, from the Sui SDK you can use
+
+```rust
+use sui::address;
+use sui::event;
+use sui::hex;
+use sui::coin::Coin;
+use sui::sui::SUI;
+```
+
+### `Singleton`
+
+The singleton is a resource type that lives in its own on-chain [object](https://docs.sui.io/concepts/object-model). In the case of this package this object owns the package's [channel](src/content/docs/dev/general-message-passing/sui/gmp-contracts#channel). The `id` of the singleton gives it a unique on-chain object identity.
+
+You can create your package's singleton as follows: 
+
+```rust
```

Comment:
I thought this was Move, not Rust?

Reply:
ya but `move` won't have the syntax highlights (its not a recognized language yet) so rust is the next best thing imo we do need the highlights

Reply:
Make sense